src/overlays/actors/ovl_En_Trt2/z_en_trt2.c

```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
+    }
+}
+
+void func_80AD4550(EnTrt2* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    u8 sp23 = func_80152498(&globalCtx->msgCtx);
+
+    if ((player->transformation != PLAYER_FORM_HUMAN) && (player->transformation != PLAYER_FORM_FIERCE_DEITY)) {
+        func_80AD3380(&this->skelAnime, sAnimations, 7);
+        this->unk_3B2 = 17;
+    }
+
+    if ((sp23 == 5) && func_80147624(globalCtx)) {
+        globalCtx->msgCtx.unk11F22 = 0x43;
+        globalCtx->msgCtx.unk12023 = 4;
+    }
+}
+
+void func_80AD4608(EnTrt2* this) {
+    s16 temp = this->unk_3B6 - 1;
+
+    if (temp != 0) {
+        this->unk_3B6 = temp;
+        return;
+    }
+
+    temp = this->unk_3B8 + 1;
+    if (temp >= 3) {
+        this->unk_3B8 = 0;
+        this->unk_3B6 = (s32)(Rand_ZeroOne() * 60.0f) + 20;
+    } else {
+        this->unk_3B8 = temp;
+        this->unk_3B6 = 1;
+    }
+}
+
+void func_80AD469C(EnTrt2* this, GlobalContext* globalCtx) {
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600FEF0, &D_06000A44, NULL, NULL, 0);
+    this->actor.draw = func_80AD56E8;
+}
+
+void func_80AD46F8(EnTrt2* this) {
+    if ((this->actor.home.pos.y + 60.0f) < this->actor.world.pos.y) {
+        this->actor.velocity.y = -0.2f;
+    } else if (this->actor.world.pos.y <= (this->actor.home.pos.y + 50.0f)) {
+        this->actor.velocity.y = 0.2f;
+    }
+}
+
+s32 func_80AD475C(EnTrt2* this, Path* path, s32 arg2) {
+    Vec3s* points;
+    s32 count;
+    f32 phi_f12;
+    s32 ret;
+    f32 phi_f14;
+    s32 arg = arg2;
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    Vec3f sp30;
+
+    points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+    count = path->count;
+    ret = false;
+    Math_Vec3s_ToVec3f(&sp30, &points[arg]);
+
+    if (arg == 0) {
+        phi_f12 = points[1].x - points[0].x;
+        phi_f14 = points[1].z - points[0].z;
+    } else if ((u32)count == (u32)(arg + 1)) {
+        phi_f12 = points[count - 1].x - points[count - 2].x;
+        phi_f14 = points[count - 1].z - points[count - 2].z;
+    } else {
+        phi_f12 = points[arg + 1].x - points[arg - 1].x;
+        phi_f14 = points[arg + 1].z - points[arg - 1].z;
+    }
+
+    func_8017B7F8(&sp30, func_80086B30(phi_f12, phi_f14) * 10430.378f, &sp44, &sp40, &sp3C);
+
+    if (((this->actor.world.pos.x * sp44) + (sp40 * this->actor.world.pos.z) + sp3C) > 0.0f) {
+        ret = true;
+    }
+    return ret;
+}
+
+s16 func_80AD48F8(Path* path, s32 arg1, Vec3f* arg2, f32* arg3) {
+    Vec3s* points;
+    f32 phi_f14;
+    f32 phi_f12;
+
+    if (path != NULL) {
+        points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+        points = &points[arg1];
+        phi_f14 = points->x - arg2->x;
+        phi_f12 = points->z - arg2->z;
+    } else {
+        phi_f14 = 0.0f;
+        phi_f12 = 0.0f;
+    }
+    *arg3 = (phi_f14 * phi_f14) + (phi_f12 * phi_f12);
+    return Math_Acot2F(phi_f12, phi_f14) * 10430.378f;
```

Comment:
```suggestion
    return RADF_TO_BINANG(Math_Acot2F(phi_f12, phi_f14));
```
Hopefully there's enough digits of accuracy for this to match